File: app/src/main/java/com/imangazalievm/bubbble/domain/userprofile/UserProfileInteractor.java
package com.imangazalievm.bubbble.domain.userprofile;


import com.imangazalievm.bubbble.domain.global.models.User;
import com.imangazalievm.bubbble.domain.global.repository.UsersRepository;
import com.imangazalievm.bubbble.presentation.mvp.global.SchedulersProvider;

import javax.inject.Inject;

import io.reactivex.Single;

public class UserProfileInteractor {

    private UsersRepository usersRepository;
    private SchedulersProvider schedulersProvider;

    @Inject
    public UserProfileInteractor(UsersRepository usersRepository,
                                 SchedulersProvider schedulersProvider) {
        this.usersRepository = usersRepository;
        this.schedulersProvider = schedulersProvider;
    }

    public Single<User> getUser(long userId) {
        return usersRepository.getUser(userId)
                .subscribeOn(schedulersProvider.io());
    }

}

File: app/src/main/java/com/imangazalievm/bubbble/domain/userprofile/UserProfileInteractor.kt
package com.imangazalievm.bubbble.domain.userprofile

import com.imangazalievm.bubbble.data.users.UsersRepository
import com.imangazalievm.bubbble.domain.global.models.User
import com.imangazalievm.bubbble.presentation.global.SchedulersProvider
import io.reactivex.Single
import javax.inject.Inject

class UserProfileInteractor @Inject constructor(
    private val usersRepository: UsersRepository
) {

    suspend fun getUser(userId: Long): User {
        return usersRepository.getUser(userId)
    }

}